File: yua-sdk/src/types/embedding.ts
// --- Embedding types (OpenAI-compatible) ---

export type EmbeddingModel =
  | "yua-embed-nano"
  | "yua-embed-small"
  | "yua-embed-large"
  | "yua-embed-kr"
  | (string & {});

export interface EmbeddingCreateParams {
  model: EmbeddingModel;
  input: string | string[];
  dimensions?: number;
  encoding_format?: "float" | "base64";
}

export interface EmbeddingObject {
  object: "embedding";
  index: number;
  embedding: number[];
}

export interface EmbeddingResponse {
  object: "list";
  model: string;
  data: EmbeddingObject[];
  usage: {
    prompt_tokens: number;
    total_tokens: number;
  };
}

File: yua-sdk-python/src/yua/types/embedding.py
from __future__ import annotations

from typing import List, Literal, Optional, Union
from pydantic import BaseModel


class EmbeddingObject(BaseModel):
    object: Literal["embedding"] = "embedding"
    index: int
    embedding: List[float]


class EmbeddingUsage(BaseModel):
    prompt_tokens: int
    total_tokens: int


class EmbeddingResponse(BaseModel):
    object: Literal["list"] = "list"
    model: str
    data: List[EmbeddingObject]
    usage: EmbeddingUsage
